software/contrib/consequencer.py

```diff
@@ -121,19 +126,26 @@ def clockTrigger():
 
             # How much randomness to add to cv1-3
             # As the randomness value gets higher, the chance of a randomly selected int being lower gets higher
+            # The output will only trigger if the randint() is <= than the probability of the step in BdProb, SnProb and HhProb respectively
             if randint(0,99) < self.randomness:
-                cv1.value(randint(0, 1))
-                cv2.value(randint(0, 1))
-                cv3.value(randint(0, 1))
+                if randint(0,9) <= int(self.BdProb[self.pattern][self.step]):
+                    cv1.value(randint(0, 1))
```

Comment:
[question] I'm having trouble following this, but it's probably because consequences is on script that I haven't really played with ( I don't own a suitable drum module). 

I'm wondering why we need to generate three random values in order to set this cv value. It looks like the final cv value itself is random (line 132), then probability of the step triggering is random (the new feature of this PR, line 131), and then there's the global random as well (on line 130).

So I'm wondering if the per-step randomness should override the global random, or be 'added' to it as it is here. Am I thinking about this right?

Reply:
Looks good!